dspace-api/src/main/java/org/dspace/content/enhancer/script/ItemEnhancerByDateScript.java

```diff
@@ -0,0 +1,277 @@
+                    counter += limit;
+                }
+
+            }
+            handler.logInfo("enhanced " + total + " items");
+            log.info("enhanced " + total + " items");
+
+        } else {
+            try {
+                this.itemService.findByIds(context, items).forEachRemaining(this::enhanceItem);
+            } catch (SQLException e) {
+                handler.logError(e.getMessage());
+            }
+            handler.logInfo("enhanced " + total + " items");
+            log.info("enhanced " + total + " items");
+        }
+    }
+
+    private SolrDocumentList searchItemsInSolr(String query, String datequeryupper, String datequerylower) throws SolrServerException, IOException {
+        SolrQuery sQuery;
+        if (Objects.nonNull(query)) {
+            sQuery = new SolrQuery(query);
+        } else {
+            sQuery = new SolrQuery("*");
+        }
+        if(Objects.nonNull(datequeryupper) && Objects.nonNull(datequerylower)){
+            sQuery.addFilterQuery("lastModified:["+datequerylower+" TO " + datequeryupper +"]");
+        } else if (Objects.nonNull(datequeryupper) ){
+            sQuery.addFilterQuery("lastModified:[* TO " + datequeryupper +"]");
+        } else if (Objects.nonNull(datequerylower) ){
+            sQuery.addFilterQuery("lastModified:[" + datequerylower + " TO *]");
+        }
+        if (Objects.nonNull(entitytype)) {
+            sQuery.addFilterQuery("search.entitytype:"+entitytype);
+        }
+        sQuery.addFilterQuery(SearchUtils.RESOURCE_TYPE_FIELD+":Item");
+        if (Objects.nonNull(collection)) {
+            sQuery.addFilterQuery("location.coll:"+UUIDUtils.toString(collection));
+        }
+        sQuery.addField(SearchUtils.RESOURCE_ID_FIELD);
+        if (max > 0) {
+            sQuery.setRows(this.max);
+        } else {
+            sQuery.setRows(Integer.MAX_VALUE);
+        }
+        sQuery.setSort("lastModified_dt",SolrQuery.ORDER.asc);
+        handler.logInfo("Query Params:" + sQuery.toString());
+        QueryResponse qResp = solrSearchCore.getSolr().query(sQuery);
+        return qResp.getResults();
+    }
+
+    private void enhanceItem(Item item) {
+        if (force) {
+            itemEnhancerService.forceEnhancement(context, item);
+        } else {
+            itemEnhancerService.enhance(context, item);
+        }
+
+        uncacheItem(item);
+    }
+
+    private void uncacheItem(Item item) {
+        try {
+            context.uncacheEntity(item);
+        } catch (SQLException e) {
+            throw new SQLRuntimeException(e);
+        }
+    }
+
+    private void assignCurrentUserInContext() throws SQLException {
+        UUID uuid = getEpersonIdentifier();
+        if (uuid != null) {
+            EPerson ePerson = EPersonServiceFactory.getInstance().getEPersonService().find(context, uuid);
+            context.setCurrentUser(ePerson);
+        }
+    }
+
+    private void assignSpecialGroupsInContext() throws SQLException {
```

Comment:
Exception 'java.sql.SQLException' is never thrown in the method.

Reply:
The Exception was adopted from the existing RelatedEntityItemEnhancer and now has been removed.